src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/RPC/SerializableWorkspaceEdit.ts

```diff
@@ -0,0 +1,46 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as vscode from 'vscode';
+import { SerializableCreateDocument } from './SerializableCreateDocument';
+import { SerializableDeleteDocument } from './SerializableDeleteDocument';
+import { convertRangeFromSerializable } from './SerializableRange';
+import { SerializableRenameDocument } from './SerializableRenameDocument';
+import { SerializableTextEdit } from './SerializableTextEdit';
+
+type SerializableDocumentChange = SerializableCreateDocument | SerializableRenameDocument | SerializableDeleteDocument;
+
+export interface SerializableWorkspaceEdit {
+    changes?: {[key: string]: Array<SerializableTextEdit>};
+    documentChanges?: Array<SerializableDocumentChange>;
+}
+
+export function convertWorkspaceEditFromSerializable(data: SerializableWorkspaceEdit): vscode.WorkspaceEdit {
+    const workspaceEdit = new vscode.WorkspaceEdit();
+    if (Array.isArray(data.documentChanges)) {
+        for (const documentChange of data.documentChanges) {
+            if (documentChange.kind === 'create') {
+                workspaceEdit.createFile(vscode.Uri.parse(documentChange.uri), documentChange.options);
+            } else if (documentChange.kind === 'rename') {
+                workspaceEdit.renameFile(vscode.Uri.parse(documentChange.oldUri), vscode.Uri.parse(documentChange.newUri), documentChange.options);
+            } else if (documentChange.kind === 'delete') {
+                workspaceEdit.deleteFile(vscode.Uri.parse(documentChange.uri), documentChange.options);
+            }
+        }
+    }
+    if (data.changes !== undefined) {
+        for (const uri in data.changes) {
+            if (!data.changes.hasOwnProperty(uri)) {
+                continue;
+            }
+            const changes = data.changes[uri];
+            for (const change of changes) {
+                const range = convertRangeFromSerializable(change.range);
+                workspaceEdit.replace(vscode.Uri.parse(uri), range, change.newText);
```

Comment:
Spoke offline. This is currently deserialized in a way that it is merging both `DocumentChanges` and `Changes`. Instead it should deserialize both `DocumentChanges` and `Changes` separately and then set both of those in a new vscode.WorkspaceEdit.